oracle-linux-image-tools/cloud/vagrant-virtualbox/image-scripts.sh

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+#
+# Cleanup and package image for the "vagrant-virtualbox" image
+#
+# Copyright (c) 1982-2020 Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at
+# https://oss.oracle.com/licenses/upl.
+#
+# Description: this module provides 2 functions:
+#   cloud::image_cleanup: cloud specific actions to cleanup the image
+#     This function is optional
+#   cloud::image_package: Package the raw image for the target cloud.
+#     This function must be defined either at cloud or cloud/distribution level
+#
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+#
+
+#######################################
+# Parameter validation
+# Globals:
+#   VAGRANT_VIRTUALBOX_CPU, VAGRANT_VIRTUALBOX_MEMORY,
+#   VAGRANT_VIRTUALBOX_EXTRA_DISK_GB
+# Arguments:
+#   None
+# Returns:
+#   None
+#######################################
+cloud::image_cleanup() {
+  [[ ${VAGRANT_VIRTUALBOX_CPU_NUM} =~ ^[0-9]*$ ]]  || error "vagrant cpu count is not numeric"
+  [[ ${VAGRANT_VIRTUALBOX_MEM_SIZE} =~ ^[0-9]*$ ]]  || error "vagrant memory is not numeric"
+  [[ ${VAGRANT_VIRTUALBOX_EXTRA_DISK_GB} =~ ^[0-9]*$ ]]  || error "vagrant disk size is not numeric"
+}
+
+#######################################
+# Cleanup actions run directly on the image
+# Globals:
+#   None
+# Arguments:
+#   root filesystem directory
+#   boot filesystem directory
+# Returns:
+#   None
+#######################################
+cloud::image_cleanup() {
+  :
+}
+
+#######################################
+# Image packaging: generate box using vagrant tool
+# Globals:
+#   VM_NAME, VAGRANT_VIRTUALBOX_CPU, VAGRANT_VIRTUALBOX_MEMORY,
+#   VAGRANT_VIRTUALBOX_EXTRA_DISK_GB
+# Arguments:
+#   None
+# Returns:
+#   None
+#######################################
+cloud::image_package() {
+  local cpu="${VAGRANT_VIRTUALBOX_CPU_NUM:-$CPU_NUM}"
+  local memory="${VAGRANT_VIRTUALBOX_MEM_SIZE:-$MEM_SIZE}"
+  # convert back to VMDK
+  local vmdk=$(grep "ovf:href" "${VM_NAME}.ovf" | sed -r -e 's/.*ovf:href="([^"]+)".*/\1/')
+  vboxmanage convertfromraw System.img --format VMDK "${vmdk}" --variant Stream
+  rm System.img
+  # re-create the OVA file
+  tar cvf "${VM_NAME}.ova" "${VM_NAME}.ovf" "${vmdk}"
+  rm "${vmdk}"
+  # Import in VirtualBox and adjust cpu/memory for the box
+  vboxmanage import "${VM_NAME}.ova" \
+    --vsys 0 --vmname "${VM_NAME}" \
+    --vsys 0 --ostype "Oracle_64" \
+    --vsys 0 --cpus $cpu \
+    --vsys 0 --memory $memory
+  rm "${VM_NAME}.ova"
+  # Add additional disk
+  if [[ -n $VAGRANT_VIRTUALBOX_EXTRA_DISK_GB ]]; then
+    local disk_size_mb=$(( ${VAGRANT_VIRTUALBOX_EXTRA_DISK_GB} * 1024 ))
+    vboxmanage createhd --filename ./extra_disk.vdi --size $disk_size_mb --format VDI --variant fixed
+    vboxmanage storageattach "${VM_NAME}" --storagectl "SATA Controller" --port 1 --device 0 --type hdd --medium ./extra_disk.vdi
```

Comment:
Can you verify that this line works on all host platforms? From memory, I seem to recall that macOS and Linux use different strings for the SATA Controller. I know I hit an issue when I created some Vagrantfiles on macOS.

Reply:
It does not. Script can only run on Linux as we loopback-mount the images at the end of the process.

Reply:
Fair enough.